provider/datagen/src/transform/cldr/displaynames/language.rs

```diff
@@ -46,22 +46,46 @@ impl IterableDataProvider<LanguageDisplayNamesV1Marker> for crate::DatagenProvid
     }
 }
 
-/// Substring used to denote alternative region names data variants for a given region. For example: "BA-alt-short", "TL-alt-variant".
-const ALT_SUBSTRING: &str = "-alt-";
+/// Substring used to denote short display names data variants for a given language. For example: "az-alt-short".
+const ALT_SHORT_SUBSTRING: &str = "-alt-short";
+/// Substring used to denote long display names data variants for a given language. For example: "az-alt-long".
+const ALT_LONG_SUBSTRING: &str = "-alt-long";
+/// Substring used to denote menu display names data variants for a given language. For example: "az-alt-menu".
+const ALT_MENU_SUBSTRING: &str = "-alt-menu";
 
 impl From<&cldr_serde::language_displaynames::Resource> for LanguageDisplayNamesV1<'static> {
     fn from(other: &cldr_serde::language_displaynames::Resource) -> Self {
         let mut names = ZeroMap::new();
+        let mut short_names = ZeroMap::new();
+        let mut long_names = ZeroMap::new();
+        let mut menu_names = ZeroMap::new();
         for lang_data_entry in other.main.0.iter() {
             for entry in lang_data_entry.1.localedisplaynames.languages.iter() {
-                let region = String::from(entry.0);
-                if !region.contains(ALT_SUBSTRING) {
+                let mut region = String::from(entry.0);
+                if region.contains(ALT_SHORT_SUBSTRING) {
+                    region.truncate(region.find(ALT_SHORT_SUBSTRING).unwrap());
+                    let key = UnvalidatedStr::from_str(&region);
+                    short_names.insert(key, entry.1.as_ref());
+                } else if region.contains(ALT_LONG_SUBSTRING) {
+                    region.truncate(region.find(ALT_LONG_SUBSTRING).unwrap());
+                    let key = UnvalidatedStr::from_str(&region);
+                    long_names.insert(key, entry.1.as_ref());
+                } else if region.contains(ALT_MENU_SUBSTRING) {
+                    region.truncate(region.find(ALT_MENU_SUBSTRING).unwrap());
+                    let key = UnvalidatedStr::from_str(&region);
+                    menu_names.insert(key, entry.1.as_ref());
+                } else {
```

Comment:
This does not gate on `ALT_SUBSTRING` anymore, which means `-alt-variant`s are now included in the `names` map. Should they be in their own map or excluded?

Reply:
Yep. Modified the PR to exclude the -alt-variants for now. There is a separate issue to handle -alt-variants. https://github.com/unicode-org/icu4x/issues/1225